vhdl_lang/src/config.rs

```diff
@@ -146,6 +199,29 @@ impl Config {
                 patterns.push(path);
             }
 
+            let mut exclude_patterns = Vec::new();
+            if let Some(opt) = lib.get("exclude") {
+                if let Some(opt) = opt.as_array() {
+                    let exclude_arr = opt;
+                    for exclude in exclude_arr.iter() {
+                        let exclude = exclude
+                            .as_str()
+                            .ok_or_else(|| format!("not a string {exclude}"))?;
+
+                        let exclude = substitute_environment_variables(exclude, &subst::Env)?;
+
+                        let path = parent.join(exclude);
+                        let path = path
+                            .to_str()
+                            .ok_or_else(|| format!("Could not convert {path:?} to string"))?
+                            .to_owned();
+                        exclude_patterns.push(path);
+                    }
+                } else {
+                    return Err(format!("excludes for library {name} is not array"));
+                }
+            }
+
```

Comment:
I think that, similar to my comment above, there is some code duplication here that could be avoided.